zschema/keys.py

```diff
@@ -30,6 +30,64 @@ def __cmp__(self, other):
             return cmp(int(self.port), int(other.port))
 
 
+class TypeFactoryFactory(object):
+    """
+    Factory for TypeFactorys (e.g. SubRecordType is a TypeFactory for
+    SubRecords).
+    """
+
+    def __init__(self, cls, args=None, kwargs=None):
+        """
+        This factory acts as a constructor for cls with the first
+        len(args) positional arguments fixed by args, and with kwargs
+        serving as the default keyword arguments.
+        Note: the positional args can be set here, or in the later call,
+        but not in both.
+        """
+        if not callable(cls):
+            raise TypeError("cls must be callable.")
+
+        if args is None:
+            args = ()
+
+        if kwargs is None:
+            kwargs = {}
+
+        if not isinstance(args, (list, tuple)):
+            raise TypeError("If present, args must be a list or a tuple.")
+
+        if not isinstance(kwargs, dict):
+            raise TypeError("If present, kwargs must be a dict.")
+
+        self.args = tuple(args)
+        # Store a shallow copy, not a reference
+        self.kwargs = dict(kwargs)
```

Comment:
Trusting that this the right thing---really don't know offhand what goes on in line 64